apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -65,6 +65,9 @@ const (
 	// cluster has gone through scale up operation.
 	NoAutomatedCleanupAnnotation = "cassandra.datastax.com/no-cleanup"
 
+	// UpdateAllowedAnnotation marks the Datacenter to allow upgrades to StatefulSets even if CassandraDatacenter object was not modified. Allowed values are "once" and "always"
+	UpdateAllowedAnnotation = "cassandra.datastax.com/allow-upgrade"
+
```

Comment:
[suggestion] I also think we should pick a better name for this annotation. `allow-upgrade` is too generic, it doesn't capture the fact that we are only considering the case when the object was not modified.

Maybe something along the lines of `allow-changes-on-operator-upgrade`?

Reply:
I renamed it to ``cassandra.datastax.com/autoupdate-spec`` which seemed to be close to what many Kubernetes "known annotations" seem to follow, such as ``rbac.authorization.kubernetes.io/autoupdate`` and ``apf.kubernetes.io/autoupdate-spec``, since their behavior is quite close to what we want.

I don't want this to be "upgrade of operator" only annotation, but instead dedicate the same behavior to all StatefulSet updates the operator might make automatically, such as removing user-made changes to the StatefulSets.